package/MDAnalysis/analysis/dssp/pydssp_numpy.py

```diff
@@ -0,0 +1,276 @@
+    # ---------
+    # v1 = vec(C_i, N_i)
+    # v2 = vec(CA_{i+1}, N_i)
+    # v3 = vec(N_i, H_i) = ?
+    # we use the assumption that all the angles are 120 degrees,
+    # and |v3| = 1.01, hence
+    # we can derive v3 = (v1/|v1| + v2/|v2|)*|v3|
+
+    # get v1 = vec(C_i, N_i)
+    vec_cn = coord[1:, 0] - coord[:-1, 2]
+    vec_cn = vec_cn / np.linalg.norm(vec_cn, axis=-1, keepdims=True)
+
+    # get v2 = vec(CA_{i+1}, N_{i})
+    vec_can = coord[1:, 0] - coord[1:, 1]
+    vec_can = vec_can / np.linalg.norm(vec_can, axis=-1, keepdims=True)
+
+    vec_nh = vec_cn + vec_can
+    vec_nh = vec_nh / np.linalg.norm(vec_nh, axis=-1, keepdims=True)
+
+    # vec_(0, H) = vec(0, N) + vec_nh
+    return coord[1:, 0] + 1.01 * vec_nh
+
+
+def get_hbond_map(
+    coord: np.ndarray,
+    cutoff: float = DEFAULT_CUTOFF,
+    margin: float = DEFAULT_MARGIN,
+    return_e: bool = False,
+) -> np.ndarray:
+    """Returns hydrogen bond map
+
+    Parameters
+    ----------
+    coord : np.ndarray
+        input coordinates in either (n, 4, 3) or (n, 5, 3) shape
+        (without or with hydrogens). If hydrogens are not present, then
+        ideal positions (see :func:_get_hydrogen_atom_positions) are used.
+    cutoff : float, optional
+        cutoff, by default DEFAULT_CUTOFF
+    margin : float, optional
+        margin, by default DEFAULT_MARGIN
+    return_e : bool, optional
+        if to return energy instead of hbond map, by default False
+
+    Returns
+    -------
+    np.ndarray
+        output hbond map or energy depending on return_e param
+
+    Raises
+    ------
+
+    ImportError
+        if module `einops` is not present
+
+    .. versionadded:: 2.8.0
+    """
+    n_atoms, n_atom_types, _ = coord.shape
+    assert n_atom_types in (
+        4,
+        5,
+    ), "Number of atoms should be 4 (N,CA,C,O) or 5 (N,CA,C,O,H)"
+
+    if n_atom_types == 4:
+        h_1 = _get_hydrogen_atom_position(coord)
+    elif n_atom_types == 5:
+        h_1 = coord[1:, 4]
+        coord = coord[:, :4]
+    else:
+        raise ValueError("Number of atoms should be 4 (N,CA,C,O) or 5 (N,CA,C,O,H)")
+    # after this:
+    # h.shape == (n_atoms, 3)
+    # coord.shape == (n_atoms, 4, 3)
+
+    # distance matrix
+    n_1, c_0, o_0 = coord[1:, 0], coord[0:-1, 2], coord[0:-1, 3]
+
+    n = n_atoms - 1
+    cmap = np.tile(c_0, (n, 1, 1))
+    omap = np.tile(o_0, (n, 1, 1))
+    nmap = np.tile(n_1, (1, 1, n)).reshape(n, n, 3)
+    hmap = np.tile(h_1, (1, 1, n)).reshape(n, n, 3)
+
+    d_on = np.linalg.norm(omap - nmap, axis=-1)
+    d_ch = np.linalg.norm(cmap - hmap, axis=-1)
+    d_oh = np.linalg.norm(omap - hmap, axis=-1)
+    d_cn = np.linalg.norm(cmap - nmap, axis=-1)
+
+    # electrostatic interaction energy
+    # e[i, j] = e(CO_i) - e(NH_j)
+    e = np.pad(
+        CONST_Q1Q2 * (1.0 / d_on + 1.0 / d_ch - 1.0 / d_oh - 1.0 / d_cn) * CONST_F,
+        [[1, 0], [0, 1]],
+    )
+
+    if return_e:
+        return e
```

Comment:
explicitly mark block as not tested

Reply:
I marked the whole function if that's ok

Reply:
see comment above — please only mark the not-tested block